Tests/MongoSwiftTests/TestUtils.swift

```diff
@@ -18,6 +18,23 @@ extension XCTestCase {
     }
 }
 
+extension MongoClient {
+    internal func serverVersion() throws -> String {
+        let buildInfo = try self.db("admin").runCommand(["buildInfo": 1])
+        guard let versionString = buildInfo["version"] as? String else {
+            throw TestError(message: "buildInfo reply missing version string: \(buildInfo)")
+        }
+        return versionString
+    }
+
+    internal func serverVersionIsInRange(_ min: String?, _ max: String?) throws -> Bool {
+        let version = try self.serverVersion()
+        if let min = min, min > version { return false }
+        if let max = max, max < version { return false }
```

Comment:
Is this robust enough for comparing version strings? I'm thinking that basic string comparison logic could be problematic. For instance, "3.6.4" should be considered smaller than "3.6.10".

It may be better to split the string on `.` to extract X.Y.Z and then compare them independently. Any non-numeric content after Z can likely be ignored (e.g. "beta", a hyphen).

Context: https://stackoverflow.com/questions/27932408/compare-two-version-strings-in-swift